airbyte-integrations/connectors/source-sendgrid/source_sendgrid/manifest.yaml

```diff
@@ -1,257 +1,987 @@
-  - $ref: "#/definitions/base_stream"
-    $parameters:
-      name: "suppression_groups"
-      primary_key: "id"
-      path: "/v3/asm/groups"
-      field_path: []
-  - $ref: "#/definitions/base_stream"
-    $parameters:
-      name: "suppression_group_members"
-      primary_key: "group_id"
-      path: "/v3/asm/suppressions"
-      field_path: []
+        type: NoPagination
+      partition_router: []
+  - type: DeclarativeStream
+    name: singlesend_stats
+    primary_key:
+      - id
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $schema: http://json-schema.org/schema#
+        properties:
+          ab_phase:
+            type: string
+          ab_variation:
+            type: string
+          aggregation:
+            type: string
+          id:
+            type: string
+          stats:
+            properties:
+              bounce_drops:
+                type: integer
+              bounces:
+                type: integer
+              clicks:
+                type: integer
+              delivered:
+                type: integer
+              invalid_emails:
+                type: integer
+              opens:
+                type: integer
+              requests:
+                type: integer
+              spam_report_drops:
+                type: integer
+              spam_reports:
+                type: integer
+              unique_clicks:
+                type: integer
+              unique_opens:
+                type: integer
+              unsubscribes:
+                type: integer
+            type: object
+        type: object
     retriever:
-      $ref: "#/definitions/base_stream/retriever"
+      type: SimpleRetriever
+      requester:
+        type: HttpRequester
+        url_base: https://api.sendgrid.com/v3
+        path: /marketing/stats/singlesends
+        http_method: GET
+        request_parameters: {}
+        request_headers: {}
+        authenticator:
+          type: BearerAuthenticator
+          api_token: "{{ config['api_key'] }}"
+        request_body_json: {}
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - results
       paginator:
-        $ref: "#/definitions/offset_paginator"
-  - $ref: "#/definitions/base_stream"
-    $parameters:
-      name: "invalid_emails"
-      primary_key: "email"
-      cursor_field: "created"
-      path: "/v3/suppression/invalid_emails"
-      field_path: []
-    incremental_sync:
-      $ref: "#/definitions/incremental_sync"
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestPath
+        page_size_option:
+          type: RequestOption
+          field_name: page_size
+          inject_into: request_parameter
+        pagination_strategy:
+          type: CursorPagination
+          page_size: 50
+          cursor_value: >-
+            {{ response['_metadata']['next'].replace('https://api.sendgrid.com',
```

Comment:
Why we should parse next cursor if we choose `RequestPath` type for `page_token_option` and the entire request URL for the subsequent request will be replaced by the cursor value?

And according to docs next value is the URL of the next page of results, so looks like we can just leave `'{{ response.get("_metadata", {}).get("next", {}) }}'`

Reply:
agreed! Thank you for looking at the docs and helping me. I found that because it was the whole URL and not jsut the path, there was no way in the builder to set it as the whole url. Is there a way to do it outside of the builder?

It uses this: 
```
page_token_option:
          type: RequestPath
```

To my knowledge there is not:
```
page_token_option:
          type: RequestUrl
```

Reply:
You can do it with builder (last paragraph) - https://docs.airbyte.com/connector-development/connector-builder-ui/pagination#cursor-pagination

<img width="1017" alt="image" src="https://github.com/airbytehq/airbyte/assets/53845333/b23e08b5-7af2-41e1-9e14-da24911875d6">

Btw, it's enough to use this (with this `cursor_value - '{{ response.get("_metadata", {}).get("next", {}) }}'`): 
```
page_token_option:
          type: RequestPath
```

Reply:
Ok, I see what's happened here.

The API is something like `https://api.sendgrid.com/v3/marketing/stats/singlesends`
`_metadata.next` is then something like `https://api.sendgrid.com/marketing/stats/singlesends?page_token=abc`
They left out the `v3` and we get a 404!

Reply:
Oh..looks like a bug from API side

Reply:
No, we are messing it up! https://github.com/airbytehq/airbyte-internal-issues/issues/2291
It works if I make the base url === host

Reply:
Fixed!